error/s2n_errno.c

```diff
@@ -314,6 +313,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_TOO_MANY_CAS, "Too many certificate authorities in trust store"); \
     ERR_ENTRY(S2N_ERR_BAD_HEX, "Could not parse malformed hex string"); \
     ERR_ENTRY(S2N_ERR_CONFIG_NULL_BEFORE_CH_CALLBACK, "Config set to NULL before client hello callback. This should not be possible outside of tests."); \
+    ERR_ENTRY(S2N_ERR_API_UNSUPPORTED, "The invoked s2n-tls API is not supported by the libcrypto") \
     /* clang-format on */
```

Comment:
Since the error message mentions "libcrypto", maybe the name should too? Idk, it does seem kind of long.
```suggestion
    ERR_ENTRY(S2N_ERR_API_UNSUPPORTED_BY_LIBCRYPTO, "The invoked s2n-tls API is not supported by the libcrypto") \
    /* clang-format on */
```

Reply:
Yeah it is longer, but it might be good since there could be other reasons why an s2n-tls API is unsupported in the future. I switched it to this for now.